File: ruby/recursive_search.rb
def recursive_search(arr, target)
  last = arr.pop
  if last == target
    return true
  elsif arr.length == 0
    return false
  else
    return recursive_search(arr,target)
  end
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: true"
  puts "=>", recursive_search([1, 2, 3], 2)

  puts

  puts "Expecting: false"
  puts "=>", recursive_search([3, 2, 1], 4)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/recursive_search.js
function recursiveSearch(arr, target) {
  last = arr.pop();
  if( last === target ){
    return true
  }
  if(arr.length === 0){
    return false
  }
  return recursiveSearch(arr, target)
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: true");
  console.log("=>", recursiveSearch([1, 2, 3], 2));

  console.log("");

  console.log("Expecting: false");
  console.log("=>", recursiveSearch([3, 2, 1], 4));
}

module.exports = recursiveSearch;

// Please add your pseudocode to this file
// And a written explanation of your solution
